pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -24,6 +24,40 @@ const (
 	PausedAnnotation = "baremetalhost.metal3.io/paused"
 )
 
+// RootDeviceHints holds the hints for specifying the storage location
+// for the root filesystem for the image.
+type RootDeviceHints struct {
+	// A device name like "/dev/vda"
+	DeviceName string `json:"devicename,omitempty"`
+
+	// A SCSI bus address like 0:0:0:0
+	HCTL string `json:"hctl,omitempty"`
+
+	// Device identifier
+	Model string `json:"model,omitempty"`
+
+	// Device vendor
+	Vendor string `json:"vendor,omitempty"`
+
+	// Disk serial number
+	Serial string `json:"serial,omitempty"`
+
+	// Size of the device in GiB
+	Size int `json:"size,omitempty"`
+
+	// Unique storage identifier
+	WWN string `json:"wwn,omitempty"`
+
+	// Unique storage identifier with the vendor extension appended
+	WWNwithExtention string `json:"wwnwithextention,omitempty"`
+
+	// Uunique vendor storage identifier
+	WWNvendorExtention string `json:"wwnvendorextention,omitempty"`
+
+	// Device rotational type
+	Rotational string `json:"rotational,omitempty"`
```

Comment:
Should be a bool

Reply:
I guess it depends on the content that the field will hold. Do we want to add just values here, like "true" ? Or do we want to support the full operator of root device hints? Like =true, !=true , etc... When looking at https://docs.openstack.org/ironic/rocky/install/advanced.html , you can see that there is a full list of logical operators that can be supported

Reply:
Done, it seems that gopher module still takes string map as input so I need to do conversion to string in code, right?